File: 167. Two Sum II - Input array is sorted/Solution.py
class Solution:
    # @param {integer[]} numbers
    # @param {integer} target
    # @return {integer[]}
    def twoSum(self, numbers, target):
        res = [-1, -1]
        if len(numbers) < 2:
            return res
            
        i = 0
        j = len(numbers) - 1
        while i < j:
            if numbers[i] + numbers[j] == target:
                res[0] = i+1
                res[1] = j+1
                return res
            elif numbers[i] + numbers[j] > target:
                j -= 1
            else:
                i += 1
        return res

File: 167. Two Sum II - Input array is sorted/Solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        vector<int> res = {-1, -1};
        
        int i = 0;
        int j = numbers.size() - 1;
        while (i < j) {
            if (numbers[i] + numbers[j] == target) {
                res[0] = i+1;
                res[1] = j+1;
                return res;
            } else if (numbers[i] + numbers[j] > target) {
                j--;
            } else {
                i++;
            }
        }
        return res;
    }
};